src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,177 @@
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        check(index);
+        node = new Node(value);
+        Node<T> current = findNodeByIndex(index);
+        if (index == 0) {
+            current.next.prev = node;
+            node.prev = null;
+            node.next = current.next;
+            head = node;
+        } else {
+            node.prev = current.prev;
+            node.next = current.next;
+            current.next.prev = node;
+            current.prev.next = node;
+        }
+        T oldItem = current.item;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        check(index);
+        Node<T> current = findNodeByIndex(index);
+        if (current.equals(head)) {
+            head = head.next;
+        } else {
+            current.prev.next = current.next;
+        }
+        if (current.equals(tail)) {
+            tail = tail.prev;
+        } else {
+            current.next.prev = current.prev;
+        }
+        size--;
+        return current.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        boolean found = false;
+        Node<T> current = head;
+        for (int i = 0; i < size; i++) {
+            if (current.item == object || current.item != null && current.item.equals(object)) {
+                if (current == head) {
+                    head = current.next;
+                    if (head != null) {
+                        head.prev = null;
+                    }
+                } else if (current == tail) {
+                    tail = current.prev;
+                    tail.next = null;
+                } else {
+                    current.prev.next = current.next;
+                    current.next.prev = current.prev;
+                }
+                found = true;
+                size--;
+                break;
+            }
+            current = current.next;
+        }
+        return found;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
+        if (size == 0) {
+            return true;
+        }
         return false;
     }
+
+    public void check(int index) throws IndexOutOfBoundsException {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        check(index);
+        Node<T> node;
+        if (index < size / 2) {
+            node = head;
+            for (int i = 0; i < index; i++) {
+                node = node.next;
+            }
+        } else {
+            node = tail;
+            for (int i = size - 1; i > index; i--) {
+                node = node.prev;
+            }
+        }
+        return node;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node prev;
```

Comment:
Use generic types for the `prev` and `next` fields to avoid raw type usage: `private Node<T> prev;` and `private Node<T> next;`.